OpenKeychain/src/main/java/org/sufficientlysecure/keychain/remote/ApiPendingIntentFactory.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2016 Dominik Schürmann <dominik@dominikschuermann.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.sufficientlysecure.keychain.remote;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+
+import org.sufficientlysecure.keychain.provider.KeychainContract;
+import org.sufficientlysecure.keychain.remote.ui.RemoteCreateAccountActivity;
+import org.sufficientlysecure.keychain.remote.ui.RemoteErrorActivity;
+import org.sufficientlysecure.keychain.remote.ui.RemoteImportKeysActivity;
+import org.sufficientlysecure.keychain.remote.ui.RemotePassphraseDialogActivity;
+import org.sufficientlysecure.keychain.remote.ui.RemoteRegisterActivity;
+import org.sufficientlysecure.keychain.remote.ui.RemoteSecurityTokenOperationActivity;
+import org.sufficientlysecure.keychain.remote.ui.RemoteSelectPubKeyActivity;
+import org.sufficientlysecure.keychain.remote.ui.SelectAllowedKeysActivity;
+import org.sufficientlysecure.keychain.remote.ui.SelectSignKeyIdActivity;
+import org.sufficientlysecure.keychain.service.input.CryptoInputParcel;
+import org.sufficientlysecure.keychain.service.input.RequiredInputParcel;
+import org.sufficientlysecure.keychain.ui.ViewKeyActivity;
+
+import java.util.ArrayList;
+
+public class ApiPendingIntentFactory {
+
+    Context mContext;
+    private Intent mPendingIntentData;
```

Comment:
Because ``mPendingIntentData`` stays the same, e.g., in ``OpenPgpService.decryptAndVerifyImpl()`` different methods of ApiPendingIntentFactory are executed always with the same ``mPendingIntentData``